FRAuth/FRAuth/WebAuthn/FRWebAuthnManager.swift

```diff
@@ -0,0 +1,226 @@
+            // `AccountManager.authorizationController(controller:didCompleteWithError:)`.
+            authController.performRequests(options: .preferImmediatelyAvailableCredentials)
+        } else {
+            // If credentials are available, presents a modal sign-in sheet.
+            // If there are no locally saved credentials, the system presents a QR code to allow signing in with a
+            // passkey from a nearby device.
+            authController.performRequests()
+        }
+
+        isPerformingModalReqest = true
+    }
+    
+    public func signUpWith(userName: String, challenge: Data, userID: String, deviceName: String?) {
+        self.deviceName = deviceName
+        let publicKeyCredentialProvider = ASAuthorizationPlatformPublicKeyCredentialProvider(relyingPartyIdentifier: domain)
+
+        // Fetch the challenge from the server. The challenge needs to be unique for each request.
+        // The userID is the identifier for the user's account.
+        //let challenge = Data()
+        let userID = Data(userID.utf8)
+        
+        let registrationRequest = publicKeyCredentialProvider.createCredentialRegistrationRequest(challenge: challenge,
+                                                                                                  name: userName, userID: userID)
+        
+        // Use only ASAuthorizationPlatformPublicKeyCredentialRegistrationRequests or
+        // ASAuthorizationSecurityKeyPublicKeyCredentialRegistrationRequests here.
+        let authController = ASAuthorizationController(authorizationRequests: [ registrationRequest ] )
+        authController.delegate = self
+        authController.presentationContextProvider = self
+    
+        authController.performRequests()
+        isPerformingModalReqest = true
+    }
+    
+    //MARK: - ASAuthorizationControllerPresentationContextProviding
+    
+    public func presentationAnchor(for controller: ASAuthorizationController) -> ASPresentationAnchor {
+        return authenticationAnchor!
+    }
+    
+    //MARK: - ASAuthorizationControllerDelegate
+    
+    public func authorizationController(controller: ASAuthorizationController, didCompleteWithAuthorization authorization: ASAuthorization) {
+        
+        switch authorization.credential {
+        case let credentialRegistration as ASAuthorizationPlatformPublicKeyCredentialRegistration:
+            FRLog.i("A new passkey was registered: \(credentialRegistration)")
+            // Verify the attestationObject and clientDataJSON with your service.
+            // The attestationObject contains the user's new public key to store and use for subsequent sign-ins.
+            let int8Arr = credentialRegistration.rawAttestationObject?.bytes.map { Int8(bitPattern: $0) }
+            let attestationObject = self.convertInt8ArrToStr(int8Arr!)
+            
+            let clientDataJSON = String(data: credentialRegistration.rawClientDataJSON, encoding: .utf8)!
```

Comment:
I would recommend using `String(decoding: data, as: UTF8.self)` to convert data to String to avoid force unwrapping . (here and in the next 2 cases)

Reply:
Good point, let me have a look on that

Reply:
fixed